packages/buidler-core/src/internal/buidler-evm/stack-traces/solidityTracer.ts

```diff
@@ -387,7 +388,10 @@ export class SolidityTracer {
     const lastInstruction = trace.bytecode.getInstruction(lastStep.pc);
 
     if (isDecodedCallTrace(trace) && !jumpedIntoFunction) {
-      if (this._hasFailedInsideTheFallbackFunction(trace)) {
+      if (
+        this._hasFailedInsideTheFallbackFunction(trace) ||
+        this._hasFailedInsideTheReceiveFunction(trace)
```

Comment:
In most places I just duplicated the logic around fallback for receive. The only exception is in the implementation of these two functions, where both use the new `_hasFailedInsideFunction`.